src/main.c (parse_opt): Change no_abs_paths_flag default value in extract mode from "false" to "true".

--- cpio/src/main.c
+++ cpio/src/main.c
@@ -241,6 +241,7 @@
 
 static char *input_archive_name = 0;
 static char *output_archive_name = 0;
+static int abs_paths_flag_changed = false;
 
 static void
 license ()
@@ -429,10 +430,12 @@ crc newc odc bin ustar tar (all-caps also recognized)"), arg);
 
     case NO_ABSOLUTE_FILENAMES_OPTION:		/* --no-absolute-filenames */
       no_abs_paths_flag = true;
+      abs_paths_flag_changed = true;
       break;
 
     case ABSOLUTE_FILENAMES_OPTION:		/* --absolute-filenames */
       no_abs_paths_flag = false;
+      abs_paths_flag_changed = true;
       break;
       
     case NO_PRESERVE_OWNER_OPTION:		/* --no-preserve-owner */
@@ -652,6 +655,8 @@ process_args (int argc, char *argv[])
       save_patterns = &argv[index];
       if (input_archive_name)
 	archive_name = input_archive_name;
+      if (!abs_paths_flag_changed)
+	no_abs_paths_flag = true;
     }
   else if (copy_function == copy_out)
     {
@@ -687,6 +692,8 @@ process_args (int argc, char *argv[])
 	archive_format = arf_binary;
       if (output_archive_name)
 	archive_name = output_archive_name;
+      if (!abs_paths_flag_changed)
+	no_abs_paths_flag = false;
     }
   else
     {
@@ -707,10 +714,10 @@ process_args (int argc, char *argv[])
       CHECK_USAGE(rename_flag, "--rename", "--pass-through");
       CHECK_USAGE(append_flag, "--append", "--pass-through");
       CHECK_USAGE(rename_batch_file, "--rename-batch-file", "--pass-through");
-      CHECK_USAGE(no_abs_paths_flag, "--no-absolute-pathnames",
-		  "--pass-through");
-      CHECK_USAGE(no_abs_paths_flag, "--absolute-pathnames",
-		  "--pass-through");
+      CHECK_USAGE((abs_paths_flag_changed && no_abs_paths_flag),
+		  "--no-absolute-pathnames", "--pass-through");
+      CHECK_USAGE((abs_paths_flag_changed && !no_abs_paths_flag),
+		  "--absolute-pathnames", "--pass-through");
       CHECK_USAGE(to_stdout_option, "--to-stdout", "--pass-through");
       
       directory_name = argv[index];
